pkg/scheduler/queue/queue.go

```diff
@@ -263,10 +274,15 @@ func (q *RequestQueue) tryDispatchRequestToQuerier(broker *queueBroker, call *ne
 	if requestSent {
 		q.queueLength.WithLabelValues(string(tenantID)).Dec()
 	} else {
-		if reqToEnqueue, ok := req.(requestToEnqueue); ok {
-			// should never error; any item previously in the queue already passed validation
-			_ = broker.enqueueRequestFront(reqToEnqueue)
-		}
+		// re-casting to same type it was enqueued as; panic would indicate a bug
+		reqToEnqueue := req.(requestToEnqueue)
+		// should never error; any item previously in the queue already passed validation
+		err := broker.enqueueRequestFront(reqToEnqueue)
+		level.Error(q.log).Log(
+			"msg", "failed to re-enqueue query request after dequeue",
```

Comment:
[nit] Might be helpful to make it clear this is not expected behaviour

```suggestion
			"msg", "failed to re-enqueue query request after dequeue, this should never happen",
```